rock/python.rb

```diff
@@ -0,0 +1,50 @@
+require 'open3'
+
+module Rock
+    # Identify the available python version, per default checks
+    # Autobuild.programs['python'] setting
+    # @param[String] python_bin Path to a custom python version
+    def self.find_python_bin(python_bin = nil)
```

Comment:
Yeah. This makes no sense. Either it finds the python bin or it does not, calling find_python_bin as an alias for File.exist? is really confusing.

Reply:
"Finding" initial means to identify the user preference on the python bin, e.g., by checking custom path, Autoproj.program
There is, however, no guarantee that the binary is available unless it is checked explicitly.

Reply:
My point is that this is very confusing:

~~~
if path = Autobuild.programs['python']
   find_python(path)
end
~~~

You need two codepaths, one that checks if the file exists if it is in programs, one that calls `find_python_bin`. This is an unnecessary (and very confusing) overload. Just the size of the `if` smells really bad.